### PR TITLE
[keycloak] Add OpenShift route

### PR DESCRIPTION
**Pull request by srueg**
To support installations on OpenShift.

Signed-off-by: Simon Rüegg <simon.ruegg@vshn.ch>

### CODE REVIEW

File: charts/keycloak/templates/route.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{{- $route := .Values.keycloak.route -}}
+{{- if $route.enabled -}}
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: {{ template "keycloak.fullname" . }}
+{{- with $route.annotations }}
```
Comment (unguiculus) [inline charts/keycloak/templates/route.yaml]: Maybe just use a `{{- with $route }}` around everything?

Comment (srueg) [reply charts/keycloak/templates/route.yaml]: My idea was to copy the style of the `ingress.yaml` but if you prefer `with`, I'll change it.

Comment (unguiculus) [reply charts/keycloak/templates/route.yaml]: I'm going to update all resources. See https://github.com/codecentric/helm-charts/pull/20/commits/1b6b1070edf0b3905823f703f2274d9b9cdaa914

But I guess ingress could be improved as well. We can leave it as is for now if you like.

Comment (srueg) [reply charts/keycloak/templates/route.yaml]: Yes, I'd prefer to leave it as is. I'm not really a friend of `with` as it hides the global scope.

**Comment by srueg** [comment]
Anything left to do here?

**Pull request merged**